src/mito2/src/flush.rs

```diff
@@ -409,9 +409,17 @@ impl RegionFlushTask {
         let action_list = RegionMetaActionList::with_action(RegionMetaAction::Edit(edit.clone()));
         // We will leak files if the manifest update fails, but we ignore them for simplicity. We can
         // add a cleanup job to remove them later.
-        self.manifest_ctx
-            .update_manifest(RegionState::Writable, action_list)
+        let version = self
+            .manifest_ctx
+            .update_manifest(
+                smallvec![RegionLeaderState::Writable, RegionLeaderState::Downgrading],
```

Comment:
Looks like only this step needs a vec of `RegionLearderState`. I think we can add a `FlushReason` variant for downgrading so we can pass `RegionLeaderState::Downgrading` to the manifest context.

Then `update_manifest()` doesn't need to take a smallvec as input. It also ensures the state is downgrading.

Reply:
Good point